### PR TITLE
tests: match request body

### PR DESCRIPTION
**Pull request by jirikuncar**
Request body must match recorded one in cassettes.

### CODE REVIEW

File: tests/test_utils.go [hunk 1]
```diff
@@ -335,7 +336,31 @@ func removeURLSecrets(u *url.URL) string {
...
+	if _, err := b.ReadFrom(r.Body); err != nil {
+		return false
+	}
+	r.Body = ioutil.NopCloser(&b)
+
+	// Body must be empty or equal
+	if strings.HasPrefix(r.Header["Content-Type"][0], "multipart/form-data") {
+		fmt.Printf("skip %s ??? %s", b.String(), i.Body)
+		return true
+	}
```
Comment (zippolyte) [inline tests/test_utils.go:358]: What does this mean ?

File: tests/test_utils.go [hunk 2]
```diff
@@ -335,7 +336,31 @@ func removeURLSecrets(u *url.URL) string {
...
+	matched := (b.String() == "" || b.String() == i.Body)
```
Comment (zippolyte) [inline tests/test_utils.go:359]: why is it a match if the body is empty and different than the recorded body ?

File: tests/test_utils.go [hunk 3]
```diff
@@ -335,7 +336,31 @@ func removeURLSecrets(u *url.URL) string {
 
 // MatchInteraction checks if the request matches a store request in the given cassette.
 func MatchInteraction(r *http.Request, i cassette.Request) bool {
-	return r.Method == i.Method && removeURLSecrets(r.URL) == i.URL
+	// Default matching on method and URL without secrets
+	if !(r.Method == i.Method && removeURLSecrets(r.URL) == i.URL) {
+		return false
+	}
+
+	if r.Body == nil {
```
Comment (bkabrda) [inline tests/test_utils.go:344]: What if `r.Body == nil`, but `i.Body != nil`? Shouldn't this condition only return true if bot `r.Body` and `i.Body` are nil at the same time?

File: tests/test_utils.go [hunk 4]
```diff
@@ -335,7 +336,31 @@ func removeURLSecrets(u *url.URL) string {
...
+		return true
```
Comment (jirikuncar) [inline tests/test_utils.go:345]: @bkabrda ❓ 
```suggestion
		return i.Body == nil
```

Comment (bkabrda) [reply tests/test_utils.go:345]: Yup, that looks right. 👍

File: tests/test_utils.go (hunk 1 above)
Comment (jirikuncar) [reply tests/test_utils.go:358]: This was a hot fix for `multipart/form-data` that are not exactly the same for some reason even when the content is the same.

File: tests/test_utils.go (hunk 2 above)
Comment (jirikuncar) [reply tests/test_utils.go:359]: It's related to `GET` requests that do not contain body.

Comment (zippolyte) [reply tests/test_utils.go:359]: But shouldn't it be equal to the recorded body `i.Body` anyway ?

**Pull request merged**